staging/src/k8s.io/kubectl/pkg/cmd/version/version.go

```diff
@@ -36,10 +38,20 @@ import (
 
 // Version is a struct for version information
 type Version struct {
-	ClientVersion *apimachineryversion.Info `json:"clientVersion,omitempty" yaml:"clientVersion,omitempty"`
+	ClientVersion KubectlVersion            `json:"clientVersion,omitempty" yaml:"clientVersion,omitempty"`
 	ServerVersion *apimachineryversion.Info `json:"serverVersion,omitempty" yaml:"serverVersion,omitempty"`
 }
 
+type KubectlVersion struct {
+	apimachineryversion.Info
+	KustomizeVersion string `json:"kustomizeVersion,omitempty" yaml:"kustomizeVersion,omitempty"`
+}
+
+// String returns the expected struct-based representation of the version with KustomizeVerison appended
+func (v KubectlVersion) String() string {
+	return strings.Replace(fmt.Sprintf("%#v", v.Info), "}", fmt.Sprintf(", KustomizeVersion:%q}", v.KustomizeVersion), 1)
```

Comment:
This huge hack is related to the real problem with this PR: the default output exposes the Go struct information from the type we are printing:

```go
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:33:37Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"darwin/arm64"}
```

Note the `version.Info` in particular. Several options come to mind, all of them ugly:
- Hack the output to append the Kustomize info like I'm doing here. But this can no longer be instantiated into the api machinery version type, if that's a thing we are trying to support.
- Let the real new struct be printed, which is very ugly: `Client Version: version.KubectlVersion{Info:version.Info{Major:"1", Minor:"24+", GitVersion:"v1.24.0-alpha.3.607+1e7c1db436e5b0-dirty", GitCommit:"1e7c1db436e5b0aaaf556b2f684a6059e53af5fb", GitTreeState:"dirty", BuildDate:"2022-03-23T21:50:55Z", GoVersion:"go1.17.6", Compiler:"gc", Platform:"darwin/arm64"}, KustomizeVersion:"v4.4.1"}
- Let the default output continue to exclude Kustomize, such that you need `-o yaml|json` to see it. This is the cleanest, but also the least intuitive/useful to our users.